### ISSUE TITLE
How about using Dokken with Kitchen?

### ISSUE THREAD

**Issue by zedtux** [EugenMayer/chef-tinc-cookbook#9]
I'm giving a try to solve the issue #8 and I first try to run the tests which requires me to install Vagrant, VirtualBox and so on, while I prefer to use Docker.

Do you have anything against using [Dokken](https://github.com/someara/kitchen-dokken) in this project? (Which works by the way, I mean tests are passing).

_I had to add an `apt-get update` in the recipe at the very beginning to get it working, which could be necessary for a fresh new machine_

**Comment by EugenMayer** [comment]
I use docker for basically anything, except chef tests - esp. due to issues with tun/tap/firewall/iptables i do full-virt tests for that cases

**Comment by zedtux** [comment]
Okay, so you want to stick with Vagrant/Virtualbox.

**Comment by EugenMayer** [comment]
well not at all just against it or something - i just think it wont work due to the network-stack limitations of docker. I am using docker extensively for basically anything else, including in production. 

e.g. i am not sure right now if a docker-container can create a tun interface at all?
Usually a container can not utiliuze IPtables either

If any of this can be solved nowdays, docker would improve / speed up testing by a extreme factor - no doubt.

So basically, prove me wrong, i beg you :)

**Comment by zedtux** [comment]
Well I meant the integration tests are passing with dokken, but I'm on your side if you'd like to test it for real, exchanging packets from one node to another one.

**Comment by EugenMayer** [comment]
@zedtux what about actually adding the docken based tests and even moving them into travis for automation, leaving the vbox based tests in the `.ktichen.yml` file to for manual runs - so run both for now.

**Comment by zedtux** [comment]
@EugenMayer that's a cool idea, let's do it.

Regarding your previous comment, thinking twice about it, and I remembered that I saw [a OpenVPN Docker image](https://github.com/kylemanna/docker-openvpn) which actually manage the tun interface, so running the tests in Docker should be possible.

There is also [this SO answer](https://stackoverflow.com/questions/34913840/best-practice-for-connecting-to-a-vpn-though-docker) I found.

**Comment by EugenMayer** [comment]
@zedtux i am already working on the .travis.yml file and also on the multi-driver .ktichen.yml

i kinda failed to run dokken due to chef license accept issues and right now looking at `kitchen-docker` - are you up for a chat in gitter or slack chef?

**Comment by zedtux** [comment]
Yes sure @EugenMayer. Send me a link and I'll join. Regarding your issue, I have a working version if you need it !

**Comment by EugenMayer** [comment]
I am in gitter and actually wrote you there ( pm ) and also in the chef slack channel https://community-slack.chef.io/ and also in gitter https://gitter.im/EugenMayer/docker-sync

**Comment by zedtux** [comment]
@EugenMayer I guess this one can be closed now 😄

**Comment by EugenMayer** [comment]
Thanks for helping!
